lib/eyeshade/balances.rb

```diff
@@ -0,0 +1,147 @@
+# typed: true
+
+module Eyeshade
+  class Balances
+    include Eyeshade::Types
+    extend T::Sig
+
+    CHANNEL = "channel"
+    REFERRAL = "owner"
+
+    attr_reader :rates,
+      :default_currency,
+      :accounts
+
+    sig { params(rates: EyeshadeObject, accounts: EyeshadeArray, transactions: EyeshadeArray, default_currency: T.nilable(String)).void }
+    def initialize(rates: {}, accounts: [], transactions: [], default_currency: nil)
+      @rates = rates
+      @accounts = []
+      @account_type_hash = {}
+      @default_currency = default_currency
+      @amount_probi = 0
+      @fees_probi = 0
+      @zero = BigDecimal("0")
+
+      accounts.each do |account|
```

Comment:
I'm actually writing a new eyeshade client that returns Account and Balance as the default response objects from the eyeshade API.  They are all immutable T::Structs which IMO is critical for a financial transactions application.